test/e2e_new/kafka_channel_upstream_test.go

```diff
@@ -0,0 +1,330 @@
+	t.Cleanup(env.Finish)
+
+	names := []string{
+		"customname",
+		"name-with-dash",
+		"name1with2numbers3",
+		"name63-01234567890123456789012345678901234567890123456789012345",
+	}
+
+	for _, name := range names {
+		env.Test(ctx, t, channel.GoesReady(name))
+	}
+}
+
+// TestSmoke_ChannelImpl
+func TestSmoke_ChannelImpl(t *testing.T) {
+	t.Parallel()
+
+	ctx, env := global.Environment()
+	t.Cleanup(env.Finish)
+
+	names := []string{
+		"customname",
+		"name-with-dash",
+		"name1with2numbers3",
+		"name63-01234567890123456789012345678901234567890123456789012345",
+	}
+
+	for _, name := range names {
+		env.Test(ctx, t, channel.ImplGoesReady(name))
+	}
+
+}
+
+// TestSmoke_ChannelWithSubscription
+func TestSmoke_ChannelWithSubscription(t *testing.T) {
+	t.Parallel()
+
+	ctx, env := global.Environment()
+	t.Cleanup(env.Finish)
+
+	channelName := "mychannel"
+
+	// Install and wait for a Ready Channel.
+	env.Prerequisite(ctx, t, channel.GoesReady(channelName))
+	chRef := ch.AsRef(channelName)
+
+	names := []string{
+		"customname",
+		"name-with-dash",
+		"name1with2numbers3",
+		"name63-01234567890123456789012345678901234567890123456789012345",
+	}
+
+	for _, name := range names {
+		env.Test(ctx, t, channel.SubscriptionGoesReady(name,
+			subscription.WithChannel(chRef),
+			subscription.WithSubscriber(nil, "http://example.com")),
+		)
+	}
+}
+
+// TestSmoke_ChannelImplWithSubscription
+func TestSmoke_ChannelImplWithSubscription(t *testing.T) {
+	t.Parallel()
+
+	ctx, env := global.Environment()
+	t.Cleanup(env.Finish)
+
+	channelName := "mychannelimpl"
+
+	// Install and wait for a Ready Channel.
+	env.Prerequisite(ctx, t, channel.ImplGoesReady(channelName))
+	chRef := chimpl.AsRef(channelName)
+
+	names := []string{
+		"customname",
+		"name-with-dash",
+		"name1with2numbers3",
+		"name63-01234567890123456789012345678901234567890123456789012345",
+	}
+
+	for _, name := range names {
+		env.Test(ctx, t, channel.SubscriptionGoesReady(name,
+			subscription.WithChannel(chRef),
+			subscription.WithSubscriber(nil, "http://example.com")),
+		)
+	}
+}
+
+/*
+TestChannelChainByUsingReplyAsSubscriber tests the following scenario:
+
+EventSource ---> (Channel ---> Subscription) x 10 ---> Sink
+
+It uses Subscription's spec.reply as spec.subscriber.
+
+This test should fail with https://github.com/knative/eventing/issues/5756 done.
+
+*/
+func TestChannelChainByUsingReplyAsSubscriber(t *testing.T) {
+	// This test fails with our KafkaChannel here since the test assumes it is ok to leave Subscription's `spec.subscriber` empty.
+	// Leaving that empty is not handled in the data channel as there won't be any `egress.Destination` in the contract resource.
+	// Dataplane won't know what to do when the destination is empty.
+	// See https://github.com/knative/eventing/issues/5756#issuecomment-1012996766
+	t.Skip("We don't allow nil `spec.subscriber` in Subscription. See https://github.com/knative/eventing/issues/5756#issuecomment-1012996766")
```

Comment:
Can we track this somewhere, please?

Reply:
I added those "skipped" tests for awareness only. 

I created https://github.com/knative-sandbox/eventing-kafka-broker/issues/1779 though